File: 源码/gradle/router/app/src/main/java/com/imooc/router/demo/MainActivity.java
package com.imooc.router.demo;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;

import com.imooc.gradle.router.runtime.Router;
import com.imooc.router.annotations.Destination;

@Destination(
        url = "router://page-home",
        description = "应用主页"
)
public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Router.INSTANCE.go(v.getContext(),
                        "router://imooc/profile?name=imooc&message=hello");
            }
        });
    }

}

File: android/AndroidProject/routerdemo/src/main/java/com/example/routerdemo/MainActivity.kt
package com.example.routerdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.imooc.router.annotations.Destination

@Destination(
    url = "router://page-home",
    description = "应用主页"
)
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}